atomate/qchem/firetasks/write_inputs.py

```diff
@@ -18,32 +27,114 @@ class WriteInputFromIOSet(FiretaskBase):
     parameters are given as keys in the dictionary.
 
     required_params:
-        molecule (Molecule): molecule
         qc_input_set (QChemDictSet or str): Either a QChemDictSet object or a string
-        name for the QChem input set (e.g., "OptSet").
+        name for the QChem input set (e.g., "OptSet"). *** Note that if the molecule is to be inherited through
+        fw_spec qc_input_set must be a string name for the QChem input set. ***
 
     optional_params:
         qchem_input_params (dict): When using a string name for QChem input set, use this as a dict
         to specify kwargs for instantiating the input set parameters. For example, if you want
         to change the DFT_rung, you should provide: {"DFT_rung": ...}.
         This setting is ignored if you provide the full object representation of a QChemDictSet
         rather than a String.
+        molecule (Molecule):
+        input_file (str): Name of the QChem input file. Defaults to mol.qin
+        write_to_dir (str): Path of the directory where the QChem input file will be written,
+        the default is to write to the current working directory
     """
 
-    required_params = ["molecule", "qchem_input_set"]
-    optional_params = ["qchem_input_params"]
+    required_params = ["qchem_input_set"]
+    optional_params = [
+        "molecule", "qchem_input_params", "input_file", "write_to_dir"
+    ]
 
     def run_task(self, fw_spec):
+        input_file = "mol.qin"
```

Comment:
You can shorten code like this to:

``input_file = self["input_file"] if "input_file" in self else "mol.qin"``

Reply:
or

``input_file = self.get("input_file", "mol.qin")``

Reply:
(note that is kind of thing occurs throughout the code)

Reply:
Thanks, I believe I fixed this everywhere.